rir/src/interpreter/interp.cpp

```diff
@@ -833,13 +920,12 @@ static unsigned serializeCounter = 0;
 // Call a RIR function. Arguments are still untouched.
 RIR_INLINE SEXP rirCall(CallContext& call, InterpreterInstance* ctx) {
     SEXP body = BODY(call.callee);
-    bool bodyPreserved = false;
+    Protect p;
```

Comment:
that thing is probably a bit slow and heavy to use in one of the hottest-path functions in the interpreter. afaik it creates some stl container....

Reply:
It just keeps an integer and calls unprotect in the destructor

Reply:
I see, still potentially slow due to the C++ obj being created...